parser.go

```diff
@@ -215,13 +215,13 @@ func (p *Parser) DecodeSegment(seg string) ([]byte, error) {
 	return encoding.DecodeString(seg)
 }
 
-// Parse parses, validates, verifies the signature and returns the parsed token.
-// keyFunc will receive the parsed token and should return the cryptographic key
-// for verifying the signature. The caller is strongly encouraged to set the
-// WithValidMethods option to validate the 'alg' claim in the token matches the
-// expected algorithm. For more details about the importance of validating the
-// 'alg' claim, see
-// https://auth0.com/blog/critical-vulnerabilities-in-json-web-token-libraries/
+// Parse parses, validates, verifies the signature, and returns the parsed token. 
+// When a keyFunc is set and a tokenString is provided, the library will not accept 
+// the 'none' type algorithm, ensuring security by default. However, it is strongly 
```

Comment:
This is not entirely correct. By default no token is rejected based on an algorithm value alone. The specified algorithms is matched against a list of available algorithms `GetSigningMethod`:
https://github.com/golang-jwt/jwt/blob/62e504c2810b67f6b97313424411cfffb25e41b0/signing_method.go#L30

After this step, the key returned in `keyFunc` is forwarded to the signing method implementation, which checks whether the type of key is suitable for that particular method or not.

In order to avoid simple mistakes, the signing method of `none` only allows keys of a very special global variable `jwt.UnsafeAllowNoneSignatureType`.

Reply:
I think I captured this, but just not that intricate detail. The main reason why I made the change here is that from the original comment it is not quite clear if the Parse function will be secure by default or not. It just mentions a rather unclear recommendation "caller is strongly encouraged to set the WithValidMethods option".

But hey, if you have a better idea for wording I'm all for it. I'm just looking to make the documentation clearer for users.

Reply:
Understood. I will try to come up with a very condensed version of what I wrote for the doc. The issue that I have with the wording "When a keyFunc is set and a tokenString is provided, the library will not accept the 'none' type algorithm, ensuring security by default" is that strictly speaking is not true. If you have a keyFunc that returns `jwt.UnsafeAllowNoneSignatureType` it will accept the 'none' type. The 'none' type does not have any special handling.